src/main/java/ee/ut/App.java

```diff
@@ -46,6 +48,33 @@ public void start(Stage stage) throws IOException {
             Files.write(resultFolder.resolve("timelineGenerator.js"), Objects.requireNonNull(App.class.getResourceAsStream("timelineGenerator.js")).readAllBytes());
         }
         scene = new Scene(loadFXML("primary"), 640, 480);
+        //Confirmation on closing
+        stage.setOnCloseRequest(evt -> {
```

Comment:
Currently this asks for confirmation even when no timeline has been started - this is unnecessary and causes an exception if the user chooses to "save", since the save button has not yet been initialized. Consider either moving this into the startTimeline method or adding a "if data != null && saveFile != null" check or something similar